sdk/Pulumi/Core/Output.cs

```diff
@@ -496,7 +496,7 @@ public Output<U> Apply<U>(Func<T, Input<U>?> func)
         /// name="func"/>. The result remains an <see cref="Output{T}"/> so that dependent resources
         /// can be properly tracked.
         /// <para/>
-        /// <paramref name="func"/> is not allowed to make resources.
+        /// <paramref name="func"/> should not be used to create resources unless necessary. Creating resources within 'func' can cause unknown behaviour.
```

Comment:
```suggestion
        /// <paramref name="func"/> should not be used to create resources unless necessary as `func` may not be run during some program executions.
```

Let's make it the same as in https://github.com/pulumi/pulumi/pull/16885.